python_modules/dagster/dagster/core/workspace/context.py

```diff
@@ -92,7 +92,7 @@ def version(self) -> Optional[str]:
 
     @property
     @abstractmethod
-    def permissions(self) -> Dict[str, bool]:
+    def permissions(self) -> Dict[str, PermissionResult]:
```

Comment:
some concern for this api change - i think mypy won't care if `PermissionResult` gets checked against directly as a bool which will always be True if its a tuple instance 

maybe `PermissionResult` should be a class impl that overrides `__bool__` / `__nonzero__` to handle checks against the object instead of the `enabled` property

Reply:
@alangenfeld does the fact that there are only 2 callsites dull your concern at all? one in this file, the other in graphql for generating the perms for dagit

Reply:
that does help, though even just raising on `__bool__` / `__nonzero__` defensively is not a big code change so i think its worth considering